### PR TITLE
feat(article): add i want to react use this article

### PR DESCRIPTION
**Pull request by DaveBitter**
Add new article on the experimental React.use

### CODE REVIEW

File: data/blog/i-want-to-react-use-this.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+title: 'I want to _React.use()_ this!'
+date: '2022-11-03'
+tags: ['react', 'frontend']
+images: ['/articles/i-want-to-react-use-this/i-want-to-react-use-this.jpg']
+summary: 'A big part of working with React.js is fetching data and displaying the results. Let’s see how the new React.use() hook can can help you!'
+authors: ['dave-bitter']
+theme: 'blue'
+---
+
+There’s quite a bit of chatter lately about an experimental hook caller `React.use()`. This will fundamentally change the way you work with [React.js](https://reactjs.org/). Let’s have a look at what it is, why you would want to `React.use()` it and how it works!
```
Comment (sanderdejong88) [inline data/blog/i-want-to-react-use-this.md]: "called"?

File: data/blog/i-want-to-react-use-this.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+## What is `React.use()`?
+
+`React.use()` is an experimental hook that offers first-class support for promises and async/await. In essence, it means that you can then await asynchronous code in the root of your components.
```
Comment (sanderdejong88) [inline data/blog/i-want-to-react-use-this.md]: Remove "then"?

File: data/blog/i-want-to-react-use-this.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+Couldn’t you already do that? Nope! Sure you could have promises in React.js for, for instance, data fetching, but you always used something like `useEffect` for this. The component, or rather function that returns some JSX, always just ran. There was no way to pause it to await some asynchronous code. Until now.
+
+## Why would I want to `React.use()` this?
+
+The ability to now await some asynchronous code will for instance have a big effect on how you load data into your component. Previously, you would have to resort to `useEffect` to fetch data. You would then have to add some state where you store the data and potentially show a spinner while it the data is being fetched. This could look something like this:
```
Comment (sanderdejong88) [inline data/blog/i-want-to-react-use-this.md]: Rename "big affect" to "big impact" because it might be confusing as `useEffect` is also in this paragraph?

Comment (sanderdejong88) [reply data/blog/i-want-to-react-use-this.md]: "while it the data" -> "while the data"?

File: data/blog/i-want-to-react-use-this.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+You can simply import the `use` hook from the React.js package. Next, you pass it a fetching function, or rather, a Promise. Finally, you assign the response to a variable. The updated example would look a bit like this:
+
+```jsx
+import React, { use } from 'react'
+
+const getData = fetch('api.com/data').then((res) => res.json())
+
+const YourComponent = () => {
+  const data = use(getData)
```
Comment (sanderdejong88) [inline data/blog/i-want-to-react-use-this.md:75]: don't you need to await it?

File: data/blog/i-want-to-react-use-this.md [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+    return 'Loading...'
+  }
+
+  if (!data) {
+    return 'No data'
+  }
+
+  return (
+    <ul>
+      {data.map((datum) => (
```
Comment (sanderdejong88) [inline data/blog/i-want-to-react-use-this.md:51]: `datum`? can we choose an ENG variable name?

File: data/blog/i-want-to-react-use-this.md [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+}
+
+export default YourComponent
+```
+
+It looks like a minor change, but let’s see what you gained.
+
+Firstly, you got rid of the `useEffect` and `useState` hooks. Even better, you got rid of any client-side code! This is now a server component which, when server-side rendering, the function will pause on the server, fetch some data and then return the “filled JSX”.
+
+Secondly, you simplified the logic. As you don’t need the before-mentioned hooks and callbacks, the code is drastically simplified. Now, this is a simple demo component, but you can imagine more complex components really benefitting from this.
```
Comment (sanderdejong88) [inline data/blog/i-want-to-react-use-this.md:97]: I would move the sentence "Now, this is a simple demo ..." to the end of the chapter (after the next paragraph)

File: data/blog/i-want-to-react-use-this.md [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+
+Finally, you could remove any of the loading state. Because of the `use` hook waited on, the rest of the code will never be executed before it’s done. Of course, you still have to handle the state where there is no data returned from the endpoint, but you can forget about any loading feedback.
```
Comment (sanderdejong88) [inline data/blog/i-want-to-react-use-this.md]: Mixed usage of "it's" and "it is"

Comment (sanderdejong88) [reply data/blog/i-want-to-react-use-this.md]: "There is" vs "There's"

File: data/blog/i-want-to-react-use-this.md (hunk 4 above)
Comment (DaveBitter) [reply data/blog/i-want-to-react-use-this.md:75]: Nope, not needed

File: data/blog/i-want-to-react-use-this.md (hunk 5 above)
Comment (DaveBitter) [reply data/blog/i-want-to-react-use-this.md:51]: Datum is the singular noun of data

File: data/blog/i-want-to-react-use-this.md (hunk 6 above)
Comment (DaveBitter) [reply data/blog/i-want-to-react-use-this.md:97]: I think it is better to do that here. It is bound to the simplification of the logic and the point that I want to make is that this is really going to help with this specific point

Comment (DaveBitter) [reply data/blog/i-want-to-react-use-this.md:97]: Do you agree?

**Review by sanderdejong88 — APPROVED** [review]

**Pull request merged**